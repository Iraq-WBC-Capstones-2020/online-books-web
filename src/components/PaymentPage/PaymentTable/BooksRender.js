import React from 'react';
export default function BooksRender({ books, changeCheck }) {
  return books.map((book, index) => {
    const { title, copies, price, id, checked } = book;

    return (
      <tr
        key={index}
        className="text-center"
        style={{ textDecoration: !checked ? 'line-through' : 'none' }}
      >
        <td>
          <input
            type="checkbox"
            onChange={(event) => changeCheck(id, event.target.checked)}
            checked={checked}
            line-through
          />
        </td>
        <td>{title}</td>
        <td>{copies}</td>
        <td>{price}</td>
      </tr>
    );
  });
}
