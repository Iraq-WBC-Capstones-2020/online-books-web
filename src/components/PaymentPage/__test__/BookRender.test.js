import React from 'react';
import renderer from 'react-test-renderer';
import BooksRender from './../PaymentTable/BooksRender';

test('BookRender snapshot', () => {
  const tree = renderer.create(<BooksRender />).toJSON();
  expect(tree).toMatchSnapshot();
});
