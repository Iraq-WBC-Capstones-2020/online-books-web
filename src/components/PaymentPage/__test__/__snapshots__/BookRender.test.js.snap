// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookRender snapshot 1`] = `
Array [
  <div
    className="container"
    data-test="container"
  >

  <tr key={index} className="text-center">
        <td>
          <input type="checkbox" />
        </td>
        <td>{title}</td>
        <td>{copies}</td>
        <td>{price}</td>
      </tr>
   
]
`;
